Index: ns-3.32/wscript
===================================================================
--- ns-3.32.orig/wscript
+++ ns-3.32/wscript
@@ -799,6 +799,31 @@ def add_examples_programs(bld):
         except OSError:
             return
 
+def add_external_programs(bld):
+    all_modules = [mod[len("ns3-"):] for mod in bld.env['NS3_ENABLED_MODULES'] + bld.env['NS3_ENABLED_CONTRIBUTED_MODULES']]
+
+    try:
+        for filename in os.listdir("externals"):
+            if filename.startswith('.') or filename == 'CVS':
+                continue
+            if os.path.isdir(os.path.join("externals", filename)):
+                obj = bld.create_ns3_program(filename, all_modules)
+                obj.path = obj.path.find_dir('externals').find_dir(filename)
+                obj.source = obj.path.ant_glob('*.cc')
+                obj.target = filename
+                obj.name = obj.target
+                obj.install_path = None
+            elif filename.endswith(".cc"):
+                name = filename[:-len(".cc")]
+                obj = bld.create_ns3_program(name, all_modules)
+                obj.path = obj.path.find_dir('externals')
+                obj.source = filename
+                obj.target = name
+                obj.name = obj.target
+                obj.install_path = None
+    except OSError:
+        return           
+            
 def add_scratch_programs(bld):
     all_modules = [mod[len("ns3-"):] for mod in bld.env['NS3_ENABLED_MODULES'] + bld.env['NS3_ENABLED_CONTRIBUTED_MODULES']]
 
@@ -963,6 +988,7 @@ def build(bld):
 
     add_examples_programs(bld)
     add_scratch_programs(bld)
+    add_external_programs(bld)
 
     if env['NS3_ENABLED_MODULES'] or env['NS3_ENABLED_CONTRIBUTED_MODULES']:
         modules = env['NS3_ENABLED_MODULES']
